module/zfs/metaslab.c

```diff
@@ -5536,6 +5537,13 @@ spa_remap_blkptr(spa_t *spa, blkptr_t *bp, spa_remap_cb_t callback, void *arg)
 	if (BP_GET_NDVAS(bp) < 1)
 		return (B_FALSE);
 
+	/*
+	 * Cloned blocks can not be remapped since BRT depends on specific
+	 * vdev id and offset in the DVA[0] for its reference counting.
+	 */
+	if (!BP_IS_METADATA(bp) && brt_maybe_exists(spa, bp))
```

Comment:
I imagine it's not a big deal, but are we worried about pinning some blocks in the indirection table because they are false positives in `brt_maybe_exists`? I guess they can only show up as a false positive if there's another true-positive nearby, which would prevent us from fully deleting the indirect mapping anyway.

Reply:
False positives seem like they shouldn't be a major problem.  It's unfortunate we can't remap BPs with a BRT entry, but there's not much we can do about that.  Thanks for running this down.